src/solvers/flattening/arrays.cpp

```diff
@@ -253,7 +253,7 @@ void arrayst::add_array_constraint(const lazy_constraintt &lazy, bool refine)
   else
   {
     // add the constraint eagerly
-    prop.l_set_to_true(convert(lazy.lazy));
+    prop.lcnf(bvt(1, convert(lazy.lazy)));
```

Comment:
This rather assumes that CNF's are cheaper than assignments, which may not always be true.  For example, the AIG back-end prefers avoiding CNF.  Is there another way this can be done?  Do we need a 'set disjunct to true / false' method?

Reply:
Also much harder to read. There are many people who don't know what "CNF" is.